lib/engine-jit/src/serialize.rs

```diff
@@ -38,9 +32,61 @@ pub struct SerializableCompilation {
 
 /// Serializable struct that is able to serialize from and to
 /// a `JITArtifactInfo`.
-#[derive(Serialize, Deserialize, MemoryUsage)]
+#[derive(Serialize, Deserialize, MemoryUsage, Archive, RkyvDeserialize, RkyvSerialize)]
 pub struct SerializableModule {
     pub compilation: SerializableCompilation,
     pub compile_info: CompileModuleInfo,
     pub data_initializers: Box<[OwnedDataInitializer]>,
 }
+
+fn to_serialize_error(err: impl std::error::Error) -> SerializeError {
+    SerializeError::Generic(format!("{}", err))
+}
+
+impl SerializableModule {
+    pub fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
+        let mut serializer = SharedSerializerAdapter::new(WriteSerializer::new(vec![]));
+        let pos = serializer
+            .serialize_value(self)
+            .map_err(to_serialize_error)? as u64;
+        let mut serialized_data = serializer.into_inner().into_inner();
+        serialized_data.extend_from_slice(&pos.to_le_bytes());
+        if cfg!(target_endian = "big") {
+            serialized_data.extend_from_slice(&[b'b']);
+        } else if cfg!(target_endian = "little") {
+            serialized_data.extend_from_slice(&[b'l']);
+        }
```

Comment:
What exactly is this information for? We unconditionally wrote something in little endian above this statement.

Reply:
Rkyv is endian-dependant. We write the endianness of the metadata so when deserializing we check the endianness of the serialized data and fail in case it varies from the host.